### DISCUSSION TITLE
how to force the task list maker to use the configured unsorted list syntax?

### DISCUSSION THREAD

**Discussion by aisbergde** [gsantner/markor#1836] [Q&A] [answered]
Regarding the action bar:
there are items to create a todo item and to create unordered list items.

In settings I can change the unordered list syntax. But I would like to use the same unordered list syntax also to create todo items.

```
* unordered item 1
* Zeile 2
- [x]  task item created by action bar, it still has - instead of *
* [ ] task items how I want to create them, using *
```

Maybe some settings could be added ore it could automatically use the settings for unordered lists?

**Comment by gsantner** [comment]
"how to force ...", what you are asking is a request for enhancement. 
Otherwise it would do already what you ask for.

**Reply by aisbergde** [reply]
OK, then it is a feature request. I was not sure if I did not find some existing ways.

**Comment by harshad1** [comment]
Looks like the behavior is correct in markdown. Should be a simple fix for wikitext

**Reply by aisbergde** [reply, marked as answer]
I use it for asciidoc files. asciidoc is not supported, but I use Markors "plain text" to get minimum support.

https://docs.asciidoctor.org/asciidoc/latest/lists/checklist/

Checklists (i.e., task lists) are unordered lists that have items marked as checked ([*] or [x]) or unchecked ([ ]).

**Comment by gsantner** [comment]
Can you switch to Wikitext/Zim format for some asciidoc files and tell how it fits? Might be that wikitext even fits as better default for .asciidoc files then just plaintext.

**Reply by aisbergde** [reply]
I tryed, and I am not so fine.

of course, best would be one additional format type "asciidoc" containing some adapted action buttons. Even if asciidoc preview is not yet supported.

I never programmed in Java. But if I would know, where in the code for the content for action buttons is defined, then maybe I could modify even not knowing Java, based on the analysis of the code for other supported format types.

Would it be possible to implement at least the format type asciidoc as a placeholder just as a copy of the "plain text" type? And then I could try to modify step by step?

**Comment by gsantner** [comment]
> Would it be possible to implement at least the format type asciidoc as a placeholder 

No. Whats there also needs maintainence. Needs bugfixes. People report bugs and want fixes. Want that and that to work. I don't add something where is no ACTIVE and RECURRING contributor and maintainer for it.

**Reply by aisbergde** [reply]
Then maybe at least for the plain text format  the behavior of the action button for check box could be adapted to use the setting for unordered list instead of a hard coded "-"?

Regarding the required active development and maintenance, I understand your point. I think I should be able to maintain the content of the action bar. But I am not able to implement or maintain formatting.

What about new format type asciidoc as experimental function? Then it could be explained, why formatting is not full implemented, but only partially.

I am ready to contribute, but my possibilities are limited. So it looks like that only full stack Java developers can contribute?

**Reply by gsantner** [reply]
>  So it looks like that only full stack Java developers can contribute?

No. What you generally needs is time, for any kind of contribution. For making small fixes or additions you probably find yourself getting it done in no time. And you are done then with it as soon it is merged.  
But adding a format, properly, and maintaining it for years it is a different category. And what happens when I just add something like Zim/Wikitext where there is like once a year a contribution -- issues pile up and there is nobody to take care of it. And new contributors? Usually zero interest in fixing or improving existing formats.

So when you really want a new format with proper syntax highlighting, with view conversion, with textaction and everything.... you mainly need this: **time and patience**. The rest, like Android or Java development is something you can learn & improve over time.

todo.txt & Markdown is there for the reason I use them personally, the reason I develop this application after all.

**Comment by harshad1** [comment]
> So it looks like that only full stack Java developers can contribute?

For reference, this is the _only_ java codebase I have ever contributed to / developed in. IMO dig in and make a PR @gsantner (and sometime I) will usually help with suggestions.

**Reply by aisbergde** [reply]
What about user configuration of action buttons? In this case no new language is required, but a user could configure which action buttons in which order he wants to see and maybe the content could be adapted. Or if the content cannot be adapted, then new action buttons could be defined, which are doing, what the user wants.

**Reply by gsantner** [reply]
> but a user could configure which action buttons in which order he wants to see

You can control the order and visibility in settings, also at plaintext which is available to all non-implemented text formats.

> and maybe the content could be adapted. 

There is a button snippets and you can have anything from a small oneliner to a whole document in that.

> action buttons could be defined, which are doing, what the user wants.

When you want to programm, please contribute code to the project. I'm not inventing a code compiler or scripting engine here.

It's not "you contribute it once" and I do all of maintenance work - for something I don't really use or care about. So really if you want it (asciidoc), please think about wether you have time for it.   **It holds, when you want proper support for a format/language/syntax bear in mind it needs proper maintenance work. Proper fixes. Considerable amount of time. It's not "add once and forget".** Users are demanding much (as you can see yourself included), and that needs somebody to take care off.

**Comment by aisbergde** [comment]
I can't customize action buttons, I created a bug. It happens on all my smartphones, where Markor is installed. I will check some tablets soon, if the issue can be reproduced there.

Regarding maintenance of a new format. I am not able to do. I have my open source projects, where I am an expert in used languages, used for this project.

I prefer to do things, where I am an expert. For examples in my comercial projects I never try to program in Java, because I know, other team members are better and faster, when programming in Java.

That means, I am not able (or willing) to contribute code to open source projects in Java. My coding quality would be poor.

Anyway I like to contribute to open source projects. Often these are bug reports, feature requests. Or I take part in the communities.

Because I can't maintain code in Java, the only thing I can contribute to the Markor project are bug reports, feature requests, taking part in discussions. If this is OK, I will do so also in the future.

**Reply by gsantner** [reply]
duplicate issue, fixed.

> Because I can't maintain code in Java, the only thing I can contribute to the Markor project are bug reports, feature requests, taking part in discussions. If this is OK, I will do so also in the future.

This is good and nice, thanks! Any good-meant contribution helps.   
But also as you know, code and feature implementations not comes magically by writing descriptions 😄 . Neither does continuous code maintainence/fixes.
